lib/routes/lnurl/payment/lnurl_payment_page.dart

```diff
@@ -56,73 +60,89 @@ class LnUrlPaymentPageState extends State<LnUrlPaymentPage> {
+  Future<void> _handleLightningPaymentLimitsResponse() async {
     final minSendableSat = widget.requestData.minSendable.toInt() ~/ 1000;
     final maxSendableSat = widget.requestData.maxSendable.toInt() ~/ 1000;
-    final effectiveMinSat = max(response.send.minSat.toInt(), minSendableSat);
-    final effectiveMaxSat = min(response.send.maxSat.toInt(), maxSendableSat);
-
-    _validateEffectiveLimits(effectiveMinSat: effectiveMinSat, effectiveMaxSat: effectiveMaxSat);
-
-    _updateFormFields(effectiveMaxSat: effectiveMaxSat);
-
-    setState(() {
-      _lightningLimits = response;
-      _loading = false;
-    });
-  }
-
-  void _validateEffectiveLimits({
-    required int effectiveMinSat,
-    required int effectiveMaxSat,
-  }) {
-    if (effectiveMaxSat < effectiveMinSat) {
-      final texts = context.texts();
-      final currencyCubit = context.read<CurrencyCubit>();
-      final currencyState = currencyCubit.state;
-
-      final isFixedAmountWithinLimits = _isFixedAmount && (effectiveMinSat == effectiveMaxSat);
-      if (!isFixedAmountWithinLimits) {
-        final effMinWithdrawableFormatted = currencyState.bitcoinCurrency.format(effectiveMinSat);
-        final effMaxWithdrawableFormatted = currencyState.bitcoinCurrency.format(effectiveMaxSat);
-        throw Exception(
-          "Payment amount is outside the allowed limits, which range from $effMinWithdrawableFormatted to $effMaxWithdrawableFormatted",
-        );
-      }
-
-      final networkLimit = currencyState.bitcoinCurrency.format(
-        effectiveMinSat,
-        includeDisplayName: true,
-      );
-      throw Exception(texts.invoice_payment_validator_error_payment_below_invoice_limit(networkLimit));
+    final effectiveMinSat = min(
+      max(_lightningLimits!.send.minSat.toInt(), minSendableSat),
+      _lightningLimits!.send.maxSat.toInt(),
+    );
+    final effectiveMaxSat = min(_lightningLimits!.send.maxSat.toInt(), maxSendableSat);
+    final errorMessage = validatePayment(
+      amountSat: minSendableSat,
+      effectiveMinSat: effectiveMinSat,
+      effectiveMaxSat: effectiveMaxSat,
+      throwError: true,
+    );
+    if (errorMessage == null) {
+      await _updateFormFields(amountSat: effectiveMaxSat);
     }
   }
 
-  void _updateFormFields({
-    required int effectiveMaxSat,
-  }) {
+  Future<void> _updateFormFields({
+    required int amountSat,
+  }) async {
     if (_isFixedAmount) {
       final currencyCubit = context.read<CurrencyCubit>();
       final currencyState = currencyCubit.state;
 
       _amountController.text = currencyState.bitcoinCurrency.format(
-        effectiveMaxSat,
+        amountSat,
         includeDisplayName: false,
       );
+      await _prepareLnUrlPayment(amountSat);
     } else if (_amountFocusNode.canRequestFocus) {
       _amountFocusNode.requestFocus();
     }
   }
 
+  Future<void> _prepareLnUrlPayment(int amountSat) async {
```

Comment:
It seems like this function is called potentially multiple times per LNURL-pay flow (unless I misread)? 
For each call to the SDK's `prepareLnurlPay()` you are making a request for an invoice, which is creating a swap for each request in the SDK receiver case. So we should limit the usage of this SDK method.

Reply:
@dangeross
> It seems like this function is called potentially multiple times per LNURL-pay flow (unless I misread)?

Yes, fees are recalculated after a valid amount input is submitted. Pre-validation is the best we can do to limit usage. We'd like to avoid creating a confirmation page & dynamically update the fee on the same screen. 

Can we pre-calculate fees in any other way without creating swaps?

Reply:
It's a bit of a chicken & egg situation, because without the invoice you won't know if there's an MRH address you can pay for lower fees. 

You can't even use `prepareSendPayment()` to get the fees because it doesn't accept a LNURL/lightning address as destination. Even if you could it would be inaccurate in the MRH case.

What are your thoughts @roeierez?

Reply:
[Add a confirmation page for non-fixed amount LNURL payments](https://github.com/breez/misty-breez/pull/214/commits/37134718e3f84c0817257f2fedb57e62ad6107c7)